controllers/secure.go

```diff
@@ -103,6 +105,17 @@ func (c *SecureContext) submitRequest(rw http.ResponseWriter, req *http.Request,
 	if contentHeader := req.Header.Get("Content-Type"); len(contentHeader) > 0 {
 		request.Header.Set("Content-Type", contentHeader)
 	}
+
+	// Get RemoteAddr from the request
+	clientIP, err := GetClientIP(req)
+	if err == nil {
+		// Set headers for requests to CF API proxy
+		request.Header.Add("X-Client-IP", clientIP)
+		request.Header.Add("X-TIC-Secret", c.Settings.TICSecret)
+	} else {
+		log.Println(err)
```

Comment:
will want to do something like
```go
 		rw.WriteHeader(http.StatusInternalServerError)
 		rw.Write([]byte("unknown error. try again"))
```

you can replace the "unknown error" part with the actual error with some context: "Unable to get Client IP: " + err.Error()